rook-ceph/examples/cluster-on-pvc.yaml

```diff
@@ -0,0 +1,223 @@
+# This file is modified from the source file below. Search for "Azure" for all the modifications.
+# Source: https://github.com/rook/rook/blob/v1.7.1/cluster/examples/kubernetes/ceph/cluster-on-pvc.yaml
+# Reference: https://github.com/rook/rook/blob/master/Documentation/ceph-cluster-crd.md
+
+#################################################################################################################
+# Define the settings for the rook-ceph cluster with common settings for a production cluster on top of cloud instances.
+# At least three nodes are required in this example. See the documentation for more details on storage settings available.
+
+# For example, to create the cluster:
+#   kubectl create -f crds.yaml -f common.yaml -f operator.yaml
+#   kubectl create -f cluster-on-pvc.yaml
+#################################################################################################################
+apiVersion: ceph.rook.io/v1
+kind: CephCluster
+metadata:
+  name: rook-ceph
+  namespace: rook-ceph # namespace:cluster
+spec:
+  # Azure: Run everything on storage nodes.
+  placement:
+    all:
+      nodeAffinity:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          nodeSelectorTerms:
+            - matchExpressions:
+                - key: "storage-node"
+                  operator: Exists
+      tolerations:
+        - key: "storage-node"
+          operator: Exists
+          effect: NoSchedule
+
+  dataDirHostPath: /var/lib/rook
+  mon:
+    # Set the number of mons to be started. Must be an odd number, and is generally recommended to be 3.
+    count: 3
+    # The mons should be on unique nodes. For production, at least 3 nodes are recommended for this reason.
+    # Mons should only be allowed on the same node for test environments where data loss is acceptable.
+    allowMultiplePerNode: false
+    # A volume claim template can be specified in which case new monitors (and
+    # monitors created during fail over) will construct a PVC based on the
+    # template for the monitor's primary storage. Changes to the template do not
+    # affect existing monitors. Log data is stored on the HostPath under
+    # dataDirHostPath. If no storage requirement is specified, a default storage
+    # size appropriate for monitor data will be used.
+    volumeClaimTemplate:
+      spec:
+        # Azure: Use Azure premium SSD for MON:
+        #        https://docs.microsoft.com/en-us/azure/virtual-machines/disks-types#premium-ssd
+        storageClassName: managed-premium
```

Comment:
better use `default` since from 1.21, default storage class would be CSI driver storage class, while `managed-premium` is not

Reply:
It is recommended to use a faster storage device for MON jobs to increase their stability.